molr-mole-core/src/main/java/io/molr/mole/core/runnable/lang/ExecutionStrategyConfiguration.java

```diff
@@ -0,0 +1,89 @@
+package io.molr.mole.core.runnable.lang;
+
+import java.text.MessageFormat;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import io.molr.commons.domain.ExecutionStrategy;
+
+public class ExecutionStrategyConfiguration {
+	
+	private final ExecutionStrategy defaultStrategy;
+	private final Set<ExecutionStrategy> allowedStrategies;
+	
+	
+	private ExecutionStrategyConfiguration() {
+		allowedStrategies = ImmutableSet.copyOf(ExecutionStrategy.values());
+		defaultStrategy = ExecutionStrategy.PAUSE_ON_ERROR;
+	}
+	
+	private ExecutionStrategyConfiguration(ExecutionStrategy defaultStrategy, Set<ExecutionStrategy> allowedStrategies) {
+		this.allowedStrategies = ImmutableSet.copyOf(allowedStrategies);
+		this.defaultStrategy = defaultStrategy;
+	}
+	
+	public ExecutionStrategy defaultStrategy() {
+		return defaultStrategy;
+	}
+
+	public Set<ExecutionStrategy> allowedStrategies() {
+		return allowedStrategies;
+	}
+
+	public static class Builder {
```

Comment:
Consider documenting somewhere how the default and the allowed Strategies interact:
* at least a default has to be present
* If allowed ones are given, then the allowed ones have to contain the default one
* If no allowed ones are given, but the default is, then the only allowed one will be the default one ...
Correct?

Consider e.g. a corresponding javadoc on the build method and on the DSL clauses ...